src/Features/Features.php

```diff
@@ -0,0 +1,304 @@
+	 *
+	 * @param string $old_value Old value.
+	 * @param string $value New value.
+	 */
+	public static function maybe_disable_features( $old_value, $value ) {
+		if ( 'yes' === $value ) {
+			return;
+		}
+
+		foreach ( self::get_features() as $feature ) {
+			self::disable( $feature );
+		}
+	}
+
+	/**
+	 * Adds the Features section to the advanced tab of WooCommerce Settings
+	 *
+	 * @param array $sections Sections.
+	 * @return array
+	 */
+	public static function add_features_section( $sections ) {
+		$features = apply_filters(
+			'woocommerce_settings_features',
+			array()
+		);
+
+		if ( empty( $features ) ) {
+			return $sections;
+		}
+
+		$sections['features'] = __( 'Features', 'woocommerce-admin' );
+		return $sections;
+	}
+
+	/**
+	 * Adds the Features settings.
+	 *
+	 * @param array  $settings Settings.
+	 * @param string $current_section Current section slug.
+	 * @return array
+	 */
+	public static function add_features_settings( $settings, $current_section ) {
+		if ( 'features' !== $current_section ) {
+			return $settings;
+		}
+
+		$features = apply_filters(
+			'woocommerce_settings_features',
+			array()
+		);
+
+		if ( empty( $features ) ) {
+			return $settings;
+		}
+
+		return array_merge(
+			array(
+				array(
+					'title' => __( 'Features', 'woocommerce-admin' ),
+					'type'  => 'title',
+					'desc'  => __( 'Start using new features that are being progressively rolled out to improve the store management experience.', 'woocommerce-admin' ),
+					'id'    => 'features_options',
+				),
+			),
+			$features,
+			array(
+				array(
+					'type' => 'sectionend',
+					'id'   => 'features_options',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Conditionally loads the beta features tracking modal.
+	 *
+	 * @param array $settings Settings.
+	 * @return array
+	 */
+	public static function maybe_load_beta_features_modal( $settings ) {
+		$tracking_enabled = get_option( 'woocommerce_allow_tracking', 'no' );
+
+		if ( 'yes' === $tracking_enabled ) {
+			return $settings;
+		}
+
+		$rtl = is_rtl() ? '.rtl' : '';
+
+		wp_enqueue_style(
+			'wc-admin-beta-features-tracking-modal',
+			Loader::get_url( "beta-features-tracking-modal/style{$rtl}", 'css' ),
+			array( 'wp-components' ),
+			Loader::get_file_version( 'css' )
+		);
+
+		wp_enqueue_script(
+			'wc-admin-beta-features-tracking-modal',
+			Loader::get_url( 'wp-admin-scripts/beta-features-tracking-modal', 'js' ),
+			array( 'wp-i18n', 'wp-element', WC_ADMIN_APP ),
+			Loader::get_file_version( 'js' ),
+			true
+		);
+
+		return $settings;
+	}
+
+	/**
+	 * Loads the required scripts on the correct pages.
+	 */
+	public static function load_scripts() {
+		if ( ! Loader::is_admin_or_embed_page() ) {
+			return;
+		}
+
+		if ( ! Loader::user_can_analytics() ) {
+			return;
+		}
+
+		$features         = self::get_features();
+		$enabled_features = array();
+		foreach ( $features as $key ) {
+			$enabled_features[ $key ] = self::exists( $key );
```

Comment:
Should this not use `self:: is_enabled( $key )`?

The previous implementation relied on the logic of the option as well

Reply:
Good call @psealock.  I think this was not the way the client-side feature flags worked as we still had flag checks on the front-end, but this actually simplifies the logic quite a bit and can't see a reason not to do this right now.

Added this in https://github.com/woocommerce/woocommerce-admin/pull/6075/commits/6cddc360298d5722cf369843e0945e94961f056e and ran through the app to make sure things weren't broken.  Looking good on my end, but a quick sanity check would be nice.